agentliveness/agent.py

```diff
@@ -175,6 +179,38 @@ def _check_manila(self):
 
         return 0
 
+    def _check_ironic(self):
+        host = self.CONF.ironic_conductor_host
+        if host is None:
+            logger.warning("please provide a ironic conductor host")
+            return 0
+        
+        ironic = ironic_client.get_client(session=self._get_session(), endpoint_type='internal', api_version='1', os_ironic_api_version='1.58')
+        try:
+            try:
+                conductor = ironic.conductor.get(self.CONF.ironic_conductor_host)
+                if not conductor.alive:
+                    logger.error("Conductor %s is not alive, commencing suicide", self.CONF.ironic_conductor_host)
+                    return 1
+            except ironic_exceptions.NotFound:
+                logger.error("Conductor %s not found, commencing suicide", self.CONF.ironic_conductor_host)
+                return 1
+            
+            drivers = ironic.driver.list()
+            ipmi = next(
+                (driver for driver in drivers if driver.name == 'ipmi'),
+                None
+            )
+            if ipmi is None or self.CONF.ironic_conductor_host not in ipmi.hosts:
+                logger.error("Conductor %s is not listed in ipmi driver, commencing suicide", self.CONF.ironic_conductor_host)
+                return 1
```

Comment:
Why specifically `ipmi` and any of the drivers?
I hope, we can get rid of ipmi rather earlier than later.

Reply:
At that time only IPMI was used but I agree

Reply:
Okay, I'll create a follow up PR.

Reply:
Follow up is #6